internal/engine/wazevo/ssa/instructions.go

```diff
@@ -1753,6 +1768,75 @@ func (i *Instruction) AsVSshr(x, amount Value, lane VecLane) *Instruction {
 	return i
 }
 
+// AsExtractlane initializes this instruction as an extract lane instruction with OpcodeExtractlane on vector.
+func (i *Instruction) AsExtractlane(x Value, index byte, lane VecLane, signed bool) *Instruction {
+	i.opcode = OpcodeExtractlane
+	i.v = x
+	// We do not have a field for signedness, but `index` is a byte,
+	// so we just encode the flag in the high bits of `u1`.
+	i.u1 = uint64(index)
+	if signed {
+		i.u1 = i.u1 | 1<<32
+	}
+	i.u2 = uint64(lane)
+	switch lane {
+	case VecLaneI8x16, VecLaneI16x8, VecLaneI32x4:
+		i.typ = TypeI32
+	case VecLaneI64x2:
+		i.typ = TypeI64
+	case VecLaneF32x4:
+		i.typ = TypeF32
+	case VecLaneF64x2:
+		i.typ = TypeF64
+	}
+	return i
+}
+
+// AsInsertlane initializes this instruction as an insert lane instruction with OpcodeInsertlane on vector.
+func (i *Instruction) AsInsertlane(x, y Value, index byte, lane VecLane, signed bool) *Instruction {
+	i.opcode = OpcodeInsertlane
+	i.v = x
+	i.v2 = y
+	i.u1 = uint64(index)
+	if signed {
+		i.u1 = 1 << 32
+	}
+	i.u2 = uint64(lane)
+	i.typ = TypeV128
+	return i
+}
+
+// AsShuffle initializes this instruction as a shuffle instruction with OpcodeShuffle on vector.
+func (i *Instruction) AsShuffle(x, y Value, lane []uint64) *Instruction {
+	i.opcode = OpcodeShuffle
+	i.v = x
+	i.v2 = y
+	// Encode the 16 bytes as 8 bytes in u1, and 8 bytes in u2.
+	i.u1 = lane[7]<<56 | lane[6]<<48 | lane[5]<<40 | lane[4]<<32 | lane[3]<<24 | lane[2]<<16 | lane[1]<<8 | lane[0]
+	i.u2 = lane[15]<<56 | lane[14]<<48 | lane[13]<<40 | lane[12]<<32 | lane[11]<<24 | lane[10]<<16 | lane[9]<<8 | lane[8]
+	i.typ = TypeV128
+	return i
```

Comment:
I didn't want to add a field just for this. I am unrolling the loop explicitly, because it's just those cases anyway. The neat thing is that this way we can just use `lfc.asLoadFpuConst128(tmp.nr(), lane1, lane2)` later in the ISA on arm64.